packages/cli/src/Server.ts

```diff
@@ -923,7 +923,19 @@ export class Server extends AbstractServer {
 					signature_method: signatureMethod,
 					// eslint-disable-next-line @typescript-eslint/naming-convention
 					hash_function(base, key) {
-						const algorithm = signatureMethod === 'HMAC-SHA1' ? 'sha1' : 'sha256';
+						let algorithm: string;
+						switch (signatureMethod) {
+							case 'HMAC-SHA256':
+								algorithm = 'sha256';
+								break;
+							case 'HMAC-SHA512':
+								algorithm = 'sha512';
+								break;
+							default:
+								algorithm = 'sha1';
```

Comment:
do you know if we really have nodes that use/support `sha1`? 
[NIST deprecated `sha1` 12 years ago](https://csrc.nist.gov/pubs/sp/800/131/a/r1/final), strongly discourages any software that uses `sha1`, and [will block purchase of all such software by government agencies after 2030](https://www.nist.gov/news-events/news/2022/12/nist-retires-sha-1-cryptographic-algorithm).

Reply:
no, I do not have such info,
OAuth1 rarely used in nodes, currently only in TwitterV1, GraphQL and HttpRequest